pkg/vault/contracts/VaultFactory.sol

```diff
@@ -64,28 +74,67 @@ contract VaultFactory is Authentication {
      * @param targetAddress Expected Vault address. The function will revert if the given salt does not deploy the
      * Vault to the target address.
      */
-    function create(bytes32 salt, address targetAddress) external authenticate {
-        if (isDisabled) {
-            revert VaultAlreadyCreated();
+    function create(
+        bytes32 salt,
+        address targetAddress,
+        bytes memory vaultCreationCode,
+        bytes memory vaultAdminCreationCode,
+        bytes memory vaultExtensionCreationCode
+    ) external authenticate {
```

Comment:
Not strictly related to this PR, but is there a reason to keep `authenticate`?

The factory should be fool-proof, meaning that it should only be able to deploy the right contracts in the right order with the right argument to the right addresses.

I think that's already (almost?) the case:
- You are storing the hashes for the creation code when the factory is deployed, which means you can only deploy the right contracts.
- The arguments are stored upon creation time. Maybe we could even make them constants so as to prevent construction time mistakes.
- The final Vault address is stored and checked against the actual deployment, so there's no possibility to deploy it elsewhere.

Am I missing something?

Reply:
It also just seems weird to need governance approval to deploy the Vault

Reply:
It seems that if you remove an authorization, someone can perform a front-run attack. For example, we deploy a contract with salt and someone intercepts it and deploys it with their arguments.

Reply:
If it should be access-free, then the salt should be saved in the constructor. Otherwise, anyone can call with any salt and the factory will no longer be able to deploy other contracts.